### PR TITLE
Authz: Allow global maintainers/owner to add Product Types

### PR DESCRIPTION
**Pull request by valentijnscholten**
We use a Product Types each external client that we build software for. Ideally some non-staff users can add new Product Types, or in our case the CICD process should be able to add new product types. Currently only staff users can add new product types. But I don't want to use a staff user for my CICD deployments or assign staff permissions to architect or project managers. I think Global Owners and Global Maintainers can basically edit all the findings / engagements / products, so I suggest to also let them add new Product Types.

This PR also extracts some repeated code into a central method. The specialty about this `Permission.Product_Type_Add` is that there is no object id to check the permissions against. Luckily @StefanFl agreed the PR was a good idea, let's hope my implementation pleases the German department of the Defect Dojo team :-)

The PR also changes the wording of `Guest` users in the docs to `Regular` users which I think is a more accurate name for these types of users.

### CODE REVIEW

File: dojo/templatetags/authorization_tags.py [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
 register = template.Library()
 
 
 @register.filter
 def has_object_permission(obj, permission):
     return user_has_permission(get_current_user(), obj, Permissions[permission])
+
+
+@register.filter
+def has_permission(permission):
```
Comment (StefanFl) [inline dojo/templatetags/authorization_tags.py:15]:
```suggestion
def has_global_permission(permission):
```

Then it fits to the other checks invented with this PR.

**Review by StefanFl — CHANGES_REQUESTED** [review]

File: dojo/templatetags/authorization_tags.py (hunk 1 above)
Comment (valentijnscholten) [reply dojo/templatetags/authorization_tags.py:15]: Yeah maybe. I chose this one as I think the other one should be called has_object_permission. But I will change it and we'll see later if we want to do a big revamp or not.

Comment (StefanFl) [reply dojo/templatetags/authorization_tags.py:15]: With the authorization for configuration we will get a `has_configuration_permission` as well.Then we will have a nice triplet. I agree, the object based permission checks should have `object` as part of their names. I would like to do it after the legacy authorization has been removed.

**Review by StefanFl — APPROVED** [review]

**Review by Maffooch — APPROVED** [review]
Good cleanup

**Review by alles-klar — APPROVED** [review]

**Pull request merged**